### PR TITLE
Feature/UI task merge

### PR DESCRIPTION
**Pull request by Anant-Raman**

### CODE REVIEW

File: app/src/main/java/com/example/mobiledger/common/extention/ViewExtention.kt [hunk 1]
```diff
@@ -62,5 +62,6 @@ fun View.changeStatusBarColor(
 
 fun String.toAmount() = "\u20B9$this"
 fun String.toPercent() = "$this \u0025"
-fun String.roundToOneDecimal(num: Float) = "%.1f".format(num)
+fun String.roundToOneDecimal(num: Float) = " %.1f".format(num)
+fun String.totalTransactionTextBuilder() = "See all $this Transactions"
```
Comment (akhilj33) [inline app/src/main/java/com/example/mobiledger/common/extention/ViewExtention.kt]: Why not add this in strings file?

File: app/src/main/java/com/example/mobiledger/presentation/home/HomeAdapter.kt [hunk 2]
```diff
@@ -8,13 +8,19 @@ import androidx.recyclerview.widget.RecyclerView
...
+import com.example.mobiledger.common.extention.totalTransactionTextBuilder
+import com.example.mobiledger.common.utils.DateUtils
 import com.example.mobiledger.common.utils.GraphUtils
 import com.example.mobiledger.databinding.*
 import com.example.mobiledger.domain.entities.TransactionEntity
 import com.example.mobiledger.domain.enums.TransactionType
 import com.github.mikephil.charting.data.PieEntry
 
-class HomeAdapter(val onDeleteItemClick: (String, Int) -> Unit, val onTransactionItemClick: (TransactionEntity) -> Unit) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+class HomeAdapter(
```
Comment (akhilj33) [inline app/src/main/java/com/example/mobiledger/presentation/home/HomeAdapter.kt:19]: space?

File: app/src/main/java/com/example/mobiledger/presentation/home/HomeViewEntity.kt [hunk 3]
```diff
@@ -5,22 +5,32 @@ import androidx.annotation.StringRes
...
+        HomeViewItem(type)
+
     data class TransactionDataRow(val data: TransactionData, val type: HomeViewType = HomeViewType.TransactionData) : HomeViewItem(type)
     object EmptyDataRow : HomeViewItem(HomeViewType.EmptyData)
 }
 
 data class MonthlyData(val incomeAmount: String, val expenseAmount: String)
 
 data class TransactionData(
+    val id: String,
```
Comment (akhilj33) [inline app/src/main/java/com/example/mobiledger/presentation/home/HomeViewEntity.kt:25]: why we are sending duplicate data? As in transaction entity already had id, name etc

Comment (Anant-Raman) [reply app/src/main/java/com/example/mobiledger/presentation/home/HomeViewEntity.kt:25]: Removed, It was added later when I merged your code.

File: app/src/main/java/com/example/mobiledger/common/extention/ViewExtention.kt (hunk 1 above)
Comment (Anant-Raman) [reply app/src/main/java/com/example/mobiledger/common/extention/ViewExtention.kt]: Done

File: app/src/main/java/com/example/mobiledger/presentation/home/HomeAdapter.kt (hunk 2 above)
Comment (Anant-Raman) [reply app/src/main/java/com/example/mobiledger/presentation/home/HomeAdapter.kt:19]: What. I have added 3 parameters.

**Review by akhilj33 — APPROVED** [review]

**Pull request merged**